### PR TITLE
Flattening Deeply Nested Data (KSQL)

### PR DESCRIPTION
**Pull request by riferrei**
This tutorial showcase how to handle records containing nested data (also known as sub-fields or complex data structures) and turn the data into a flatten version of the record.

This PR addresses the following issue: https://github.com/confluentinc/kafka-tutorials/issues/190

Sincerely,

-- Ricardo

### CODE REVIEW

File: _data/tutorials.yaml [hunk 1]
```diff
@@ -186,3 +186,14 @@ udf:
     ksql: enabled
     kstreams: disabled
     kafka: disabled
+
+flatten-nested-data:
+  title: "How to flatten deeply nested events"
+  meta-description: "How to flatten deeply nested events"
+  slug: "/flatten-nested-data"
+  problem: "you have events in a Kafka topic, and each event has nested data and you want to transform the event into a dataset that is simpler to handle using a flatten structure."
```
Comment (colinhicks) [inline _data/tutorials.yaml]:
```suggestion
  problem: "you have events with nested data in a Kafka topic, and you want to transform each event into a flattened dataset that is simpler to handle."
```

File: _data/tutorials.yaml [hunk 2]
```diff
@@ -186,3 +186,14 @@ udf:
...
+  introduction: "Consider a topic containing orders and each order contains data about the customer and the product, specified as a nested data. In this tutorial, we'll write a program that transforms each order into a new version of it that contains all the data as flat fields."
```
Comment (colinhicks) [inline _data/tutorials.yaml]:
```suggestion
  introduction: "Consider a topic containing product orders. Each order contains data about the customer and the product, specified as nested data. In this tutorial, we'll write a program that transforms each order into a new version of it that contains all the data as flat fields."
```

File: _includes/tutorials/flatten-nested-data/ksql/markup/dev/create-inputs.adoc [hunk 3]
```diff
@@ -0,0 +1,6 @@
+Create the following stream that contains the representation of the order.
```
Comment (colinhicks) [inline _includes/tutorials/flatten-nested-data/ksql/markup/dev/create-inputs.adoc]:
```suggestion
Create the following stream with the representation of the orders.
```

File: _includes/tutorials/flatten-nested-data/ksql/markup/dev/create-inputs.adoc [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+We will the `STRUCT` keyword to define the fields that contains nested data.
```
Comment (colinhicks) [inline _includes/tutorials/flatten-nested-data/ksql/markup/dev/create-inputs.adoc]:
```suggestion
We use the `STRUCT` keyword to define the fields that contain nested data.
```

File: _includes/tutorials/flatten-nested-data/ksql/markup/dev/create-outputs.adoc [hunk 5]
```diff
@@ -0,0 +1,7 @@
+Note that now each field is being shown in a flat structure.
+This means that our query is working propertly.
```
Comment (colinhicks) [inline _includes/tutorials/flatten-nested-data/ksql/markup/dev/create-outputs.adoc]:
```suggestion
This means that our query is working properly.
```

File: _includes/tutorials/flatten-nested-data/ksql/markup/dev/set-properties.adoc [hunk 6]
```diff
@@ -0,0 +1,5 @@
+Now that you have stream with some events in it, let's start to leverage them. The first thing to do is set the following properties to ensure that you're reading from the beginning of the stream:
```
Comment (colinhicks) [inline _includes/tutorials/flatten-nested-data/ksql/markup/dev/set-properties.adoc]:
```suggestion
Now that you have a stream with some events in it, let's start to leverage them. The first thing to do is set the following properties to ensure that you're reading from the beginning of the stream:
```

File: _includes/tutorials/flatten-nested-data/ksql/markup/dev/transient-query.adoc [hunk 7]
```diff
@@ -0,0 +1,11 @@
+Now that our continuous query has been created, we can query the result stream called `FLATTENED_ORDERS` with a much simpler query that doesn't need to handle nested data.
```
Comment (colinhicks) [inline _includes/tutorials/flatten-nested-data/ksql/markup/dev/transient-query.adoc]:
```suggestion
We can query the new result stream called `FLATTENED_ORDERS` with a much simpler query that doesn't need to handle nested data.
```

**Comment by riferrei** [comment]
> Looks great, @riferrei. I pointed out a few things in the text.
> 
> Also I wonder about using a real Logitech product name in the example data. Not sure, but we might need to add a trademark symbol to the first reference of it. Alternatively, we could use a more generic product name for the example.

Thanks @colinhicks. And you're absolutely right: keeping references of these products would put us in a tough spot. I just committed a new version of the code with the complete removal of it.

**Pull request merged**